[Flaky CI]: test_metrics.py::test_graceful_shutdown

### Flakiness Type

Assertion failure

### Name of Test

tests/integration/test_metrics.py::test_graceful_shutdown

### Link to Test Run

https://github.com/getsentry/relay/actions/runs/8372303917/job/22923014899?pr=3295

### Details

```
=================================== FAILURES ===================================
____________________________ test_graceful_shutdown ____________________________
[gw1] linux -- Python 3.10.13 /home/runner/work/relay/relay/.venv/bin/python3
Traceback (most recent call last):
  File "/home/runner/work/relay/relay/tests/integration/test_metrics.py", line 1389, in test_graceful_shutdown
    with pytest.raises(requests.ConnectionError):
  File "/home/runner/work/relay/relay/.venv/lib/python3.10/site-packages/_pytest/python_api.py", line 966, in __exit__
    fail(self.message)
  File "/home/runner/work/relay/relay/.venv/lib/python3.10/site-packages/_pytest/outcomes.py", line 196, in fail
    raise Failed(msg=reason, pytrace=pytrace)
Failed: DID NOT RAISE <class 'requests.exceptions.ConnectionError'>
```

--- a/tests/integration/test_metrics.py
+++ b/tests/integration/test_metrics.py
@@ -1268,6 +1268,8 @@ def test_graceful_shutdown(mini_sentry, relay):
     relay.send_metrics(project_id, metrics_payload)
     relay.process.send_signal(signal.SIGTERM)
 
+    time.sleep(0.1)
+
     # Try to send another metric (will be rejected)
     metrics_payload = f"transactions/now:666|c|T{timestamp}"
     with pytest.raises(requests.ConnectionError):

